backends/cxxrtl/runtime/cxxrtl/cxxrtl.h

```diff
@@ -837,6 +837,29 @@ std::ostream &operator<<(std::ostream &os, const value_formatted<Bits> &vf)
 	return os;
 }
 
+// An object that can be passed to a `commit()` method in order to produce a replay log of every
+// state change in the simulation.
+struct observer {
+	// Called when a `commit()` method for a wire is about to update the `chunks` chunks at `base`
+	// with `chunks` chunks at `value` that have a different bit pattern. It is guaranteed that
+	// `chunks` is equal to the wire chunk count and `base` points to the first chunk.
+	virtual void on_commit(size_t chunks, const chunk_t *base, const chunk_t *value) = 0;
```

Comment:
So this would be a pointer to `wire` except that the wire type is parametrized, so we take `chunk_t *` to the wire's data instead?

Reply:
(Regarding the `base` argument)

Reply:
Correct. In practice there are two reasons this argument is there:
- It provides the identity of the wire (which can be correlated via debug items to its name, etc.)
- It provides the value before the commit.

Reply:
I guess as an option we could pass in the number of bits instead of number of chunks, providing more infromation. This is a trade-off, I assume the number of bits can be looked up through debug items anyway, and passing in the number of chunks optimizes this more towards fast recording of state.

Reply:
Yes. This interface is solely concerned with the storage representation.